### PR TITLE
Fix fs memory leak when concurrently getting fs in JobPath

### PR DESCRIPTION
**Pull request by bradyoo**
This code depends on https://github.com/Alluxio/alluxio/pull/10700

### CODE REVIEW

**Comment by bradyoo** [comment]
@gpang

**Comment by bradyoo** [comment]
rebased

File: job/server/src/main/java/alluxio/job/plan/transform/format/ReadWriterUtils.java [hunk 1]
```diff
@@ -51,7 +49,6 @@ public static Configuration readNoCacheConf() {
     Configuration conf = new Configuration();
     conf.setEnum(PropertyKey.USER_FILE_READ_TYPE_DEFAULT.getName(), ReadType.NO_CACHE);
     // The cached filesystem might not be configured with the above read type.
```
Comment (gpang) [inline job/server/src/main/java/alluxio/job/plan/transform/format/ReadWriterUtils.java]: should this comment be removed?

File: job/server/src/main/java/alluxio/job/plan/transform/format/JobPath.java [hunk 2]
```diff
@@ -93,20 +96,28 @@ public JobPath(String scheme, String authority, String path) {
...
+
+    FileSystemKey key = new FileSystemKey(this, conf);
+    FileSystem fileSystem = CACHE.get(key);
+    if (fileSystem != null) {
+      return fileSystem;
+    }
+
+    fileSystem = super.getFileSystem(conf);
+
+    synchronized (CACHE) {
```
Comment (gpang) [inline job/server/src/main/java/alluxio/job/plan/transform/format/JobPath.java]: I don't think you will need `synchronized`. Instead there should be a `putIfAbsent` which may be applicable, or a `compute` method.

**Pull request closed without merging**